lib/KCheckbox/index.vue

```diff
@@ -104,19 +112,40 @@
         default: true,
       },
       /**
-       * Checked state
+       * Reactive v-model checkbox state. Updates with the `input` event.
        */
-      checked: {
-        type: Boolean,
+      /*
+       * The reactive state of the checkbox which is used with v-model.
+       * It is changed with the "input" event.
+       * If used as an array, "value" prop is added/removed from it based on the checkbox state.
+       * If used as a boolean, it is set to true when checked and false when unchecked.
+       * If used as any other type, it is set to "value" prop when checked and null when unchecked.
+       */
+      inputValue: {
+        type: [Array, Boolean, Number, String, Object],
         default: false,
       },
       /**
-       * Indeterminate state. Overrides `checked` state
+       * Indeterminate state. Overrides `v-model`.
        */
       indeterminate: {
         type: Boolean,
         default: false,
       },
+      /**
+       * Checkbox value
+       */
```

Comment:
I think it'd be great if we specify here that this is the value that will be set by the `v-model` when the checkbox is checked.